### ISSUE TITLE
[Feature] More robust user lighting settings

### ISSUE THREAD

**Issue by JBeastCreations** [Teardown-Issue-Tracker-Maintainers/Teardown-Issue-Tracker#344]
### Is your feature request related to a problem (if not, explain why you think this feature should be added)? Please describe.

I have a relatively powerful GPU all things considered, and while I find myself pretty CPU-bound in Teardown, I have a fair deal of GPU headroom that's going underutilized even as there are ways that I feel the game could be pushed visually.

For example, (assuming I remember the stream correctly and it is still unchanged), ambient occlusion rays only extend ~30 meters. While that is quite a distance in the Teardown campaign, in many mods significant light-leaking can occur. The same is true for specular occlusion, where light can visibly leak through buildings. Volumetric fog can also display significant noise and artifact, as do reflections. 

### Describe the solution you'd like

I would like a series of graphical options that are more extensive than a simple low-medium-high preset. The ability to control maximum ray length for various effects and/or samples per-pixel (a simple slider would do). By and large, just an ability to scale Teardown upwards for next-gen cards. 

### Describe alternatives you've considered

As I stated in a feature request I wrote for an enhanced photo mode, I was looking for any sort of graphical enhancement mods that may be available, or any way to access the renderer's buffers at all, but none that I see are available. If an API that could access rendering information would be released and modders could tinker with its' capabilities, I believe that would also be a wonderful option. 

### Additional context

![image](https://github.com/Teardown-Issue-Tracker-Maintainers/Teardown-Issue-Tracker/assets/79931729/5b2a3621-e499-40b8-9868-39a4f1da38c6)
A shot taken looking at a display screen on the Apple Store 5th avenue map.

As you can see, even with screen-space information clearly present, and a space deep below-ground, light-leaking is significant and unavoidable.
![image](https://github.com/Teardown-Issue-Tracker-Maintainers/Teardown-Issue-Tracker/assets/79931729/9f023206-18d9-4f77-bc79-4e121aabd845)

To reduce light-leaking requires moving all the way to the other end of the screen, and even then it is not entirely removed.

**Comment by gaberundlett-tuxedo** [comment]
I agree it would be nice to expose these settings. You're right, for both reflections and AO, the maximum trace distance is hard-coded as 32 meters and 24 meters respectively.
I'll track this internally, maybe we can do something about exposing them.

**Comment by JBeastCreations** [comment]
I am very grateful you are taking this under consideration! I've felt for a while now that (without the hard data to back this up, of course) more advanced settings in the shadow volume could support much greater scalability and quality without much more than a handful of sliders.

While I think ray length and SPP are the most important wins in terms of scalability and quality, I think there are other good avenues for improving lighting: supporting multiple SPPs for transparent materials to eliminate the dithering artifacts, implementing the GBuffer supercover tracing for other effects like shadows to cut down on light leaking, etc. Nonetheless, thank you so much for taking my post under consideration at all!
